src/main/java/org/wildfly/security/auth/realm/ldap/UserPasswordCredentialLoader.java

```diff
@@ -113,8 +117,9 @@ public SupportLevel getCredentialAcquireSupport(final Class<? extends Credential
 
         @Override
         public SupportLevel getEvidenceVerifySupport(final Class<? extends Evidence> evidenceType, final String algorithmName) throws RealmUnavailableException {
-            // If it is obtainable, it is verifiable.
-            return getCredentialAcquireSupport(PasswordCredential.class, null);
+            // If we have a password guess and the credential is obtainable, we are able to verify.
+            if ( ! PasswordGuessEvidence.class.isAssignableFrom(evidenceType)) return SupportLevel.UNSUPPORTED;
+            return getCredentialAcquireSupport(PasswordCredential.class, algorithmName);
```

Comment:
I think that if any password credential algorithm is supported, then password guessing is supported.  Thus the check should be for an algorithm name of "null", especially because the PasswordGuessEvidence does not support algorithm names.  Otherwise this looks correct to me.